src/lib/sedlexing.mli

```diff
@@ -48,13 +48,16 @@ val create: (Uchar.t array -> int -> int -> int) -> lexbuf
         position [pos], and return the number of characters provided. A
         return value of 0 means end of input. *)
 
-val set_curr_p: lexbuf -> Lexing.position -> unit
-    (** set initial [lex_curr_p] for [lexbuf],
-        if set to [Lexing.dummy_pos], Sedlexing would not track position info for you *)
+val set_position: lexbuf -> Lexing.position -> unit
+    (** set the initial tracked input position for [lexbuf].
+        If set to [Lexing.dummy_pos], Sedlexing will not track position
+        information for you. *)
 
 val set_filename: lexbuf -> string -> unit
-    (** [set_filename lexbuf file] set [lex_curr_p.pos_fname] for [lexbuf],
-        this would be used in [lexing_positions] and [with_tokenizer] *)
+    (** [set_filename lexbuf file] sets the filename known to [lexbuf].
+        This is also made available through the [lexing_positions] and
+        [with_tokenizer] calls, where it is exposed via the returned
+        [Lexing.position.pos_fname] values. *)
```

Comment:
These sentences are *weird*. What about `... sets the filename to [file] in [lexbuf]. This also sets the `Lexing.pos_fname` field in the position returned by ....` ?

Reply:
That sounds fine. How is:

`    (** [set_filename lexbuf file] sets the filename to [file] in
        [lexbuf]. It also sets the {!Lexing.pos_fname} field in the
        {!Lexing.position} records returned by [lexing_positions] and
        the function returned by [with_tokenizer]. *)`

The last bit seems a bit awkward but I'm not sure what to do.

Reply:
So maybe

`    (** [set_filename lexbuf file] sets the filename to [file] in
        [lexbuf]. It also sets the {!Lexing.pos_fname} field in
        returned {!Lexing.position} records. *)`

Reply:
Good for me.